typescript/sdk/src/token/deploy.ts

```diff
@@ -65,6 +59,51 @@ export class HypERC20Deployer extends GasRouterDeployer<
     }); // factories not used in deploy
   }
 
+  routerContractName<K extends keyof HypERC20Factories>(
+    config: ERC20RouterConfig,
+  ): K {
+    if (isCollateralConfig(config)) {
+      return (
+        isFastConfig(config) ? TokenType.fastCollateral : TokenType.collateral
+      ) as K;
+    } else if (isSyntheticConfig(config)) {
+      return (
+        isFastConfig(config) ? TokenType.fastSynthetic : TokenType.synthetic
+      ) as K;
+    } else {
+      throw new Error('Unknown collateral type when constructing router name');
+    }
+  }
+
+  async constructorArgs(_: ChainName, config: ERC20RouterConfig): Promise<any> {
+    if (isCollateralConfig(config)) {
+      return [config.token, config.mailbox];
+    } else if (isNativeConfig(config)) {
+      return config.scale ? [config.scale, config.mailbox] : [config.mailbox];
+    } else if (isSyntheticConfig(config)) {
+      return [config.decimals, config.mailbox];
+    } else {
+      throw new Error('Unknown collateral type when constructing arguments');
+    }
+  }
```

Comment:
I think you can strongly type this if you infer the `factory` type from the `config.tokenType`